addons/mass_mailing_event/models/event_registration.py

```diff
@@ -9,4 +9,7 @@ class EventRegistration(models.Model):
     _mailing_enabled = True
 
     def _mailing_get_default_domain(self, mailing):
```

Comment:
I'd suggest that instead of modifying this method, we modify `_compute_mailing_domain` in `MassMailing` add a `elif` statement like so:
```py
    def _compute_mailing_domain(self):
        for mailing in self:
            if not mailing.mailing_model_id:
                mailing.mailing_domain = ''
            elif mailing.mailing_filter_id:
                mailing.mailing_domain = mailing.mailing_filter_id.mailing_domain
            elif self.env.context.get('default_mailing_domain'):
                mailing.mailing_domain = self.env.context['default_mailing_domain']
            else:
                mailing.mailing_domain = repr(mailing._get_default_mailing_domain())
```
That way, we preserve the domain specified in the original action :)

Reply:
+1, we should not rely only on active_model but (try to) use default domain. Maybe in combination of active_model versus mailing model to be sure.

Reply:
Thanks for the suggestion, I agree with the proposed change
Note: I had to disable `compute_sudo` to avoid losing the context in the compute

Reply:
@agr-odoo Hello, does `compute_sudo` really have an impact on the context ? When I'm trying locally, I get the correct context even with `compute_sudo=True`, what is the issue you encountered ?

Also @tde-banana-odoo suggested that you add a check for this condition: verify that the `default_mailing_model_id` in the context is the same as the `mailing.mailing_model_id`.

@tde-banana-odoo Asking for confirmation for the previous statement here, you mentioned `active_model`, but in this case it is `event.event` while the mailing_model_id is `event.registration`, not sure how they can be checked against another if that's what you really meant.

Reply:
@abd-msyukyu-odoo 
I realized now it's an issue only in 17 where we clean the context when making the sudo call
https://github.com/odoo/odoo/blob/3ae269683e0c814a6a0bfeaace6098e2f09cefcc/odoo/api.py#L568
But, it's not needed in 16, so I'll address this is the fw port

Also, imo TDE meant that in case you really need to use `active_model` then you should also check the extra condition, so not in this case, but I may be wrong ^^

Reply:
Hello @agr-odoo I discussed with Thibault and we think it's best to add the check that `mailing.mailing_model_id` is the same as the `context.default_mailing_model_id` to the condition.

After that it should be ready for a merge :)

Cheers!